tests/NULS/SignerTests.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "HexCoding.h"
+#include "PrivateKey.h"
+#include <gtest/gtest.h>
+
+#include "NULS/Address.h"
+#include "NULS/Signer.h"
+#include "NULS/TransactionBuilder.h"
+
+using namespace TW;
+using namespace TW::NULS;
+
+inline auto makeInput(std::string priKey, std::string addressFrom, std::string addressTo,
+                      uint256_t amount, uint32_t chainId, uint32_t assetId, std::string nonce, std::string remark,
+                      uint256_t balance) {
+    Proto::SigningInput input;
+    input.set_private_key(priKey);
+
+    input.set_from_address(addressFrom);
+    input.set_to_address(addressTo);
+
+    Data amountData = store(amount);
+    std::string amountStr;
+    amountStr.insert(amountStr.begin(), amountData.begin(), amountData.end());
+    input.set_amount(amountStr);
+
+    input.set_chain_id(chainId);
+    input.set_idassets_id(assetId);
+    input.set_nonce(nonce);
+    input.set_remark(remark);
+
+    Data balanceData = store(balance);
+    std::string balanceStr;
+    balanceStr.insert(balanceStr.begin(), balanceData.begin(), balanceData.end());
+    input.set_balance(balanceStr);
+
+    input.set_timestamp(0x5d8885f8);
+    return input;
+}
+
+TEST(NULSSigner, Sign) {
+    auto input = makeInput("9ce21dad67e0f0af2599b41b515a7f7018059418bab892a7b68f283d489abc4b",
+                           "NULSd6Hgj7ZoVgsPN9ybB4C1N2TbvkgLc8Z9H", "NULSd6Hgied7ym6qMEfVzZanMaa9qeqA6TZSe",
+                           10000000, 1, 1, "0000000000000000", "", 100000000);
+    auto signer = NULS::Signer(input);
+    std::vector<uint8_t> signature = signer.sign();
```

Comment:
```suggestion
    Data signature = signer.sign();
```
you can use `Data` typealias